modules/openapi-generator/src/main/resources/elm/DateOnly.mustache

```diff
@@ -24,11 +25,8 @@ dateOnlyEncoder model =
 decodeIsoString : String -> Decoder DateOnly
 decodeIsoString str =
     case fromIsoString str of
-        Just date ->
+        Result.Ok date ->
             Decode.succeed date
 
-        Nothing ->
-            Decode.fail <|
-                "Cannot convert "
-                    ++ str
-                    ++ " to DateOnly"
+        Result.Err msg ->
+            Decode.fail msg
```

Comment:
Something like `Decode.fail <| "Cannot decode ISO string: " ++ msg` makes clear that it's the decoder that failed

Reply:
The error message in the new date-extra is

> Failed to create a Date from string 'foo': Invalid ISO 8601 format

Do you still want to add `"Cannot decode ISO string: "`?

Reply:
Well the thing is this is a `Decoder` error, not a `Date` error. But the type already indicates that so I think we can stick with the `Decode.fail msg`.